src/util/QuadUtil.ts

```diff
@@ -0,0 +1,63 @@
+import type { Readable } from 'stream';
+import arrayifyStream from 'arrayify-stream';
+import { DataFactory, StreamParser, StreamWriter } from 'n3';
+import type { Literal, NamedNode, Quad } from 'rdf-js';
+import streamifyArray from 'streamify-array';
+import { RepresentationMetadata } from '../ldp/representation/RepresentationMetadata';
+import { TEXT_TURTLE } from './ContentTypes';
+import { pipeSafe } from './StreamUtil';
+import { LDP, RDF } from './UriConstants';
+import { toNamedNode } from './UriUtil';
+
+/**
+ * Generates a quad with the given subject/predicate/object and pushes it to the given array.
+ */
+export const pushQuad =
+  (quads: Quad[], subject: NamedNode, predicate: NamedNode, object: NamedNode | Literal): number =>
+    quads.push(DataFactory.quad(subject, predicate, object));
+
+/**
+ * Helper function to generate type quads for a Container or Resource.
+ * @param subject - Subject for the new quads.
+ * @param isContainer - If the identifier corresponds to a container.
+ *
+ * @returns The generated quads.
+ */
+export const generateResourceQuads = (subject: NamedNode, isContainer: boolean): Quad[] => {
```

Comment:
This function and the next might actually be a `ContainerUtil` (with multiple implementations at some point). It's not about quads in general.